### PR TITLE
Update develop

### LINKED ISSUE #1: Weatherdata update wiith weatheradapter service
# Weatherdata update wiith weatheradapter service

- [x] Used Weatheradapter Service and delete mapping ressource_ids and weather_adapter

### PR DESCRIPTION
**Pull request by mlabadie**
* Update get_data with weatheradapter fonction

    * I created in weatherDatasource an endpoint function that returns the enpoint associated to the parameter name of 
    WeatherDataSource and docstring documentation. 
    This function is then used in get_data to automatically define the endpoint parameter
    * I add check_forecast function to check if endpoint correspond to forecast or not 
    * Then i update get_data function with forecast condition to use weatheradapter (agroservice) or weatheradapter_forecast according to endpoint correspond to forecast or not

### CODE REVIEW

File: src/weatherdata/ipm.py [hunk 1]
```diff
@@ -11,30 +11,30 @@
...
-                 'Metos':'metos'}
-
-weather_adapter = {'fmi': ipm.get_weatheradapter_fmi,
-                   'yr': ipm.get_weatheradapter_yr,
-                   'lmt':'',
-                   'meteobot':'',
-                   'fw':'',
-                   'metos':''
-                    }
+#ipm = IPM()
```
Comment (pradal) [inline src/weatherdata/ipm.py]: It is easier to comment with triple quotes : """
Thus you just have to add/rm two lines to manage the "dead" code

File: src/weatherdata/ipm.py [hunk 2]
```diff
@@ -11,30 +11,30 @@
 import pandas
```
Comment (pradal) [inline src/weatherdata/ipm.py:11]: first import external modules (pandas)
Then agroservices

File: src/weatherdata/ipm.py [hunk 3]
```diff
@@ -78,9 +78,26 @@ def list_available_parameters(self):
...
+        """
+        endpoints = self.ipm.weatheradapter_service()
+
+        if self.name in endpoints:
+            endpoint = endpoints[self.name]
         
+        return endpoint    
          
-    def get_data(self,parameters, station_id, daterange):
+    def get_data(self,parameters=[1002,3002], station_id=101104, daterange=pandas.date_range('2020-06-12T00:00:00','2020-07-03T00:00:00',freq='H',tz="UTC")):
```
Comment (pradal) [inline src/weatherdata/ipm.py]: - Put the def of the function on multiple line.
- Simplify the date range.
- Where these parameters come from? Can they be obtain by query?

File: src/weatherdata/__init__.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+from . import ipm
+from .ipm import *
+
+
+# sub packages inside bioservices.
+
+#import mapping
+#from . import apps
+
```
Comment (pradal) [inline src/weatherdata/__init__.py]: Remove empty spaces and comments

File: src/weatherdata/ipm.py [hunk 5]
```diff
@@ -59,7 +59,7 @@ def get_station_ids(self):
         data = data[["name","id","coordinates"]]
         return data
 
-    def list_available_parameters(self):
+    def get_list_available_parameters(self):
```
Comment (pradal) [inline src/weatherdata/ipm.py]: replace by get_parameters

File: src/weatherdata/ipm.py [hunk 6]
```diff
@@ -141,28 +242,21 @@ def list_ressources(self):
         ---------
             dictionnary with name and description of available weatherdatasource on IPM service
         """
-        rep = self.ws.get_weatherdatasource()
+        rep = self.ipm.get_weatherdatasource()
         return {item['name']:item['description'] for item in rep}
-
+        
     def get_ressource(self, name):
```
Comment (pradal) [inline src/weatherdata/ipm.py:269]: get_ressource() -> resource()

File: src/weatherdata/ipm.py [hunk 7]
```diff
@@ -141,28 +242,21 @@ def list_ressources(self):
...
-for names in values:
-    if 'features' in values[names]:
-        station_ids[names]=[values[names]['features'][item]['properties'] for item in range(len(values[names]['features']))]
-        coord[names]=[values[names]['features'][item]['geometry'] for item in range(len(values[names]['features']))]
-
-    else:
-        station_ids[names]= 'no stations for this ressources'
-        coord[names]= 'no stations for this ressources'
\ No newline at end of file
+            raise NotImplementedError()
```
Comment (pradal) [inline src/weatherdata/ipm.py]: Add new line at the end of the file

**Review by pradal — CHANGES_REQUESTED** [review]

File: src/weatherdata/ipm.py (hunk 6 above)
Comment (mlabadie) [reply src/weatherdata/ipm.py:269]: How to differenciate list_resources and get_resources if i replace get_resource by resource() and list_resources by resources also

**Comment by mlabadie** [comment]
I made a mistake in the commits
6134ecc is an update of IPM taking into account the reviews you made and the second 2466f43 is really about the __init__.py file on which I removed spaces and comments

**Review by pradal — APPROVED** [review]

**Pull request merged**